### PR TITLE
Love command checks

### PR DESCRIPTION
**Pull request by ChrisLovering**
## Description

This makes a few changes to the `.love` command to make it slightly less creepy, more consistent with other lovefest commands and also makes it give consistent output, regardless of input order.

The changes I have made are:
 - Added and `in_month` check, to ensure the command can only run during February (valentine season)
 - Require that both members have opted into the love fest event (by checking roles)
 - Sort the members to be checked before hashing, to ensure symmetrical results.
 - Removed some now un-used checks since we don't accept arbitrary strings as input

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

File: bot/exts/valentines/lovecalculator.py [hunk 1]
```diff
@@ -22,45 +24,41 @@
...
-            else:
-                # Otherwise normalise case and remove any leading/trailing whitespace
-                arg = arg.strip().title()
-            # This has to be done manually to be applied to usernames
-            return clean_content(escape_markdown=True).convert(ctx, arg)
+        if not all((
+            Lovefest.role_id in [role.id for role in who.roles],
+            Lovefest.role_id in [role.id for role in whom.roles]
+        )):
+            raise BadArgument("Both members must have the love fest role!")
```
Comment (Akarys42) [inline bot/exts/valentines/lovecalculator.py:54]:
```suggestion
            raise BadArgument("Both members must have the lovefest role!")
```

**Review by Akarys42 — CHANGES_REQUESTED** [review]
Sorry 😔

File: bot/exts/valentines/lovecalculator.py (hunk 1 above)
Comment (ChrisLovering) [reply bot/exts/valentines/lovecalculator.py:54]: How about this?
```suggestion
            raise BadArgument("Both members must have the `Lovefest` role!")
```

Comment (ChrisLovering) [reply bot/exts/valentines/lovecalculator.py:54]: actually, never mind. It's referred to as lovefest in other commands, so lets stay consistent.

File: bot/exts/valentines/lovecalculator.py [hunk 2]
```diff
@@ -22,45 +24,41 @@
...
-            else:
-                # Otherwise normalise case and remove any leading/trailing whitespace
-                arg = arg.strip().title()
-            # This has to be done manually to be applied to usernames
-            return clean_content(escape_markdown=True).convert(ctx, arg)
+        if not all((
+            Lovefest.role_id in [role.id for role in who.roles],
+            Lovefest.role_id in [role.id for role in whom.roles]
+        )):
+            raise BadArgument("Both members must have the lovefest role!")
```
Comment (wookie184) [inline bot/exts/valentines/lovecalculator.py:54]: Might be worth mentioning how to get the lovefest role, something like
```suggestion
            raise BadArgument("Both members must have the lovefest role! You can get this by running the `.lovefest sub` command.")
```
if the `who` or `whom` is the invoker we probably wouldn't need to require the lovefest role, but the current behaviour is fine.

**Review by wookie184 — CHANGES_REQUESTED** [review]
Could add a footer to the embed sent saying you can unsubscribe with `.lovefest unsub`, but you could also not do that, I don't' mind.

**Review by wookie184 — APPROVED** [review]
lgtm

**Review by Xithrius — APPROVED** [review]
Yes.

**Pull request merged**